Python nightly failing due to Jinja2 import from collections.abc

Error:
```
ImportError while loading conftest '/home/vsts/work/1/s/sphinx_gallery/tests/conftest.py'.
sphinx_gallery/tests/conftest.py:16: in <module>
    from sphinx.application import Sphinx
../../../.local/lib/python3.10/site-packages/sphinx/application.py:32: in <module>
    from sphinx.config import Config
../../../.local/lib/python3.10/site-packages/sphinx/config.py:27: in <module>
    from sphinx.util.tags import Tags
../../../.local/lib/python3.10/site-packages/sphinx/util/tags.py:11: in <module>
    from jinja2 import nodes
/usr/lib/python3/dist-packages/jinja2/__init__.py:33: in <module>
    from jinja2.environment import Environment, Template
/usr/lib/python3/dist-packages/jinja2/environment.py:15: in <module>
    from jinja2 import nodes
/usr/lib/python3/dist-packages/jinja2/nodes.py:19: in <module>
    from jinja2.utils import Markup
/usr/lib/python3/dist-packages/jinja2/utils.py:647: in <module>
    from markupsafe import Markup, escape, soft_unicode
/usr/lib/python3/dist-packages/markupsafe/__init__.py:13: in <module>
    from collections import Mapping
E   ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)
```

Fixed here: https://github.com/pallets/jinja/pull/867, included in version Jinja2>=2.10.2

Sphinx only requires [Jinja2>=2.3](https://github.com/sphinx-doc/sphinx/blob/a1489099ec35202de7509fc0ea9c7f67563f6cdf/setup.py#L24)

~Should we make a Sphinx issue or wait~ we could specify Jinja2 version in our CI, WDYT @larsoner ?

--- a/continuous_integration/azure/install.sh
+++ b/continuous_integration/azure/install.sh
@@ -55,7 +55,7 @@ elif [ "$DISTRIB" == "nightly" ]; then
     # pip install --no-use-pep517 -q https://api.github.com/repos/matplotlib/matplotlib/zipball/master
     #
     # So for now we'll just live without NumPy.
-    pip install -q --upgrade --pre sphinx joblib pytest-cov pygments colorama
+    pip install -q --upgrade --pre sphinx joblib pytest-cov pygments colorama jinja2>=2.3
     pip install -q .
 elif [ "$DISTRIB" == "minimal" ]; then
     python -m pip install --upgrade . pytest pytest-cov coverage

